packages/react-data-grid-addons/src/cells/headerCells/filters/AutoCompleteFilter.js

```diff
@@ -38,14 +38,16 @@ class AutoCompleteFilter extends React.Component {
         continue;
       }
 
-      let filterTermValue = filterTerms[key].value;
+      if (typeof columnValue !== 'undefined' && typeof filterTerms[key].value !== 'undefined') {
+        let filterTermValue = filterTerms[key].value.toString();
 
-      let checkValueIndex = columnValue.trim().toLowerCase().indexOf(filterTermValue.trim().toLowerCase());
-      let columnMatchesSearch = checkValueIndex !== -1 && (checkValueIndex !== 0 || columnValue === filterTermValue);
+        let checkValueIndex = columnValue.toString().trim().toLowerCase().indexOf(filterTermValue.trim().toLowerCase());
```

Comment:
call `columnValue.toString()` only once as this can change. put this on a var named `columnValueText` for example.